core/app/models/spree/payment.rb

```diff
@@ -241,5 +246,28 @@ def update_order
         end
       end
 
+      def create_eligible_credit_event
+        # When cancelling an order, a payment with the negative amount
+        # of the payment total is created to refund the customer. That
+        # payment has a source of itself (Spree::Payment) no matter the
+        # type of payment getting refunded, hence the additional check
+        # if the source is a store credit.
+        return unless store_credit? && source.is_a?(Spree::StoreCredit)
+
+        # creates the store credit event
+        source.update_attributes!({
+          action: Spree::StoreCredit::ELIGIBLE_ACTION,
```

Comment:
Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.